Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -0,0 +1,133 @@
+// Sources/SwiftProtobuf/BinaryTypeAdditions.swift - Per-type binary coding
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to the proto types defined in ProtobufTypes.swift to provide
+/// type-specific binary coding and decoding.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Binary encoding and decoding methods for messages.
+public extension Message {
+  /// Returns a `Data` value containing the Protocol Buffer binary format
+  /// serialization of the message.
+  ///
+  /// - Parameters:
+  ///   - partial: Binary serialization requires all `required` fields be
+  ///     present; if `false` (the defaul), this method throws
+  ///     `BinaryEncodingError.missingRequiredFields` if any were missing. When
```

Comment:
Suggest:  ... if `false` (the default), this method will call `Message.isInitialized` before encoding to verify that all required fields are present.  The `Message.isInitialized` method will throw `BinaryEncodingError.missingRequiredFields` if it finds any required fields are not set.